language/tools/move-cli/src/utils/credential.rs

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) The Diem Core Contributors
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{bail, Context, Result};
+use std::fs;
+use toml_edit::easy::Value;
+
+#[derive(Clone)]
+pub struct TestMode {
+    pub test_path: String,
+}
+
+pub fn get_move_home_path(test_mode: Option<TestMode>) -> String {
+    let mut move_home;
+    if let Some(test_mode) = test_mode {
+        move_home = std::env::var("TEST_MOVE_HOME").unwrap();
+        if !test_mode.test_path.is_empty() {
+            move_home.push_str(&test_mode.test_path);
+        }
+    } else {
+        move_home = std::env::var("MOVE_HOME").unwrap_or_else(|_| {
+            format!(
+                "{}/.move",
+                std::env::var("HOME").expect("env var 'HOME' must be set")
+            )
+        });
+    }
+    move_home
+}
+
+pub fn get_credential_path(test_mode: Option<TestMode>) -> String {
+    get_move_home_path(test_mode) + "/credential.toml"
```

Comment:
Would you mind moving `credential.toml` to a constant and perhaps renaming it to something more descriptive like `movey_api_key.toml`?